crates/kubelet/src/kubelet.rs

```diff
@@ -214,7 +214,7 @@ pub trait Provider {
         _namespace: String,
         _pod: String,
         _container: String,
-    ) -> Result<Vec<String>, failure::Error> {
+    ) -> Result<Vec<u8>, failure::Error> {
         Err(NotImplementedError {}.into())
```

Comment:
Is there a reason we default to this? It seems we should require that the implementing provider implement this method. If they choose to implement it with a `NotImplementedError` than that's their decision.

Reply:
Right now it serves as more of a reminder to us. I’ll create an issue to change this before we release

Reply:
Leaving it like this means we need to change two crates in the future instead of making the trait definition how it should be from the start.

Reply:
As it is out of scope in this PR, I'll do it in a follow up

Reply:
I've thought about this some more and since implementing logs and exec is completely optional, I think this should remain as is

Reply:
Should it return an empty Vec then? This will cause other work flows to error out.